streams/src/test/java/org/apache/kafka/streams/perf/YahooBenchmark.java

```diff
@@ -0,0 +1,375 @@
+                    return true;
+                }
+            }
+            System.out.println("Initializing topic " + topic);
+
+            parent.processedRecords.set(0);
+            parent.processedBytes = 0;
+            Properties props = new Properties();
+            props.put(ProducerConfig.CLIENT_ID_CONFIG, clientId);
+            props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, parent.kafka);
+            props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+            props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class);
+
+            KafkaProducer<String, byte[]> producer = new KafkaProducer<>(props);
+
+            long startTime = System.currentTimeMillis();
+
+            ProjectedEvent event = new ProjectedEvent();
+            event.userID = UUID.randomUUID().toString(); // not used
+            event.pageID = UUID.randomUUID().toString(); // not used
+            event.addType = "banner78";  // not used
+            event.ipAddress = "1.2.3.4"; // not used
+
+            Map<String, Object> serdeProps = new HashMap<>();
+            final Serializer<ProjectedEvent> projectedEventSerializer = new JsonPOJOSerializer<>();
+            serdeProps.put("JsonPOJOClass", ProjectedEvent.class);
+            projectedEventSerializer.configure(serdeProps, false);
+
+            for (int i = 0; i < numRecords; i++) {
+                event.eventType = eventTypes[rand.nextInt(eventTypes.length - 1)];
+                event.adID = ads[rand.nextInt(ads.length - 1)];
+                event.eventTime = System.currentTimeMillis();
+                byte[] value = projectedEventSerializer.serialize(topic, event);
+                producer.send(new ProducerRecord<>(topic, event.adID, value));
+                parent.processedRecords.getAndIncrement();
+                parent.processedBytes += value.length + event.adID.length();
+            }
+            producer.close();
+
+            long endTime = System.currentTimeMillis();
+
+
+            parent.printResults("Producer Performance [records/latency/rec-sec/MB-sec write]: ", endTime - startTime);
+            return true;
+        }
+        return false;
+    }
+
+
+    public void run() throws Exception {
+        int numCampaigns = 100;
+        int adsPerCampaign = 10;
+
+        String[] ads = new String[numCampaigns * adsPerCampaign];
+        if (maybeSetupPhaseCampaigns(campaignsTopic, "simple-benchmark-produce-campaigns", false,
+            numCampaigns, adsPerCampaign, ads)) {
+            maybeSetupPhaseEvents(eventsTopic, "simple-benchmark-produce-events", false,
+                parent.numRecords, ads);
+            return;
+        }
+
+        CountDownLatch latch = new CountDownLatch(1);
+        Properties props = parent.setStreamProperties("simple-benchmark-yahoo" + new Random().nextInt());
+        props.put(StreamsConfig.NUM_STREAM_THREADS_CONFIG, Runtime.getRuntime().availableProcessors());
```

Comment:
Do we really want to do this? Might it be better to have a config for this? Or just run it with a fixed number of threads